.github/workflows/ci.yml

```diff
@@ -608,6 +608,24 @@ jobs:
           testing/bin/ptl suite run \
             --suite ${{ matrix.suite }} --config config-${{ matrix.config }} --bind=off --logs-dir logs/ --timeout 2h \
             --trino-jdk-version zulu_${{ matrix.jdk }}
+      - name: Generate Jacoco report
+        run: |
+          export TRINO_VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)
+          mkdir -p /tmp/trino-jacoco/reports
+          mkdir -p "core/trino-server/target/trino-server-${TRINO_VERSION}-trino-jars"
+          # copy only trino JARs, since we don't care about coverage of external dependencies
+          find "core/trino-server/target/trino-server-${TRINO_VERSION}-hardlinks" -name 'trino-*.jar' -print0 | xargs --null cp --link --no-clobber -t "core/trino-server/target/trino-server-${TRINO_VERSION}-trino-jars"
+          java -jar testing/trino-product-tests-launcher/target/org.jacoco.cli-0.8.6-nodeps.jar \
+            report /tmp/trino-jacoco/presto-master/*.exec \
+            --classfiles "core/trino-server/target/trino-server-${TRINO_VERSION}-trino-jars" \
+            --name "coverage pt (${{ matrix.config }}, ${{ matrix.suite }})" \
+            --csv /tmp/trino-jacoco/reports/report.csv
+      - name: Upload Jacoco report
+        uses: actions/upload-artifact@v2
+        with:
+          name: Jacoco report for (${{ matrix.config }}, ${{ matrix.suite }})
```

Comment:
Could this be included in the `Upload test report` step? Both these artifacts should be relatively small, and I think having fewer artifacts would be easier to manage.